<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QgsAuthSslTestDialog</class>
 <widget class="QDialog" name="QgsAuthSslTestDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>617</width>
    <height>710</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Custom Certificate Configuration</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLabel" name="lblWarningIcon">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>48</width>
         <height>48</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>48</width>
         <height>48</height>
        </size>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="lblWarningText">
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Save a custom SSL server configuration, importing certificate from server or file. WARNING: Only save configurations when necessary.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="childrenCollapsible">
      <bool>false</bool>
     </property>
     <widget class="QgsCollapsibleGroupBoxBasic" name="grpbxServer">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>Import certificate</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <property name="spacing">
        <number>6</number>
       </property>
       <property name="leftMargin">
        <number>6</number>
       </property>
       <property name="topMargin">
        <number>6</number>
       </property>
       <property name="rightMargin">
        <number>6</number>
       </property>
       <property name="bottomMargin">
        <number>6</number>
       </property>
       <item>
        <widget class="QRadioButton" name="radioServerImport">
         <property name="text">
          <string>From server</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">btngrpImport</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="frameServerImport">
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <property name="spacing">
           <number>6</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <property name="spacing">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="lblHttps">
              <property name="text">
               <string>https://</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="leServer">
              <property name="minimumSize">
               <size>
                <width>175</width>
                <height>0</height>
               </size>
              </property>
              <property name="placeholderText">
               <string>www.example.com</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="spinbxPort">
              <property name="maximum">
               <number>50000</number>
              </property>
              <property name="value">
               <number>443</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QToolButton" name="btnConnect">
            <property name="font">
             <font>
              <pointsize>11</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Connect</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lblTimeout">
            <property name="font">
             <font>
              <pointsize>11</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Timeout</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinbxTimeout">
            <property name="font">
             <font>
              <pointsize>11</pointsize>
             </font>
            </property>
            <property name="suffix">
             <string> sec</string>
            </property>
            <property name="minimum">
             <number>5</number>
            </property>
            <property name="maximum">
             <number>90</number>
            </property>
            <property name="value">
             <number>15</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radioFileImport">
         <property name="text">
          <string>From file</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">btngrpImport</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="frameFileImport">
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <property name="spacing">
           <number>0</number>
          </property>
          <item>
           <widget class="QLineEdit" name="leCertPath">
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="placeholderText">
             <string>PEM/DER formatted file</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="btnCertPath">
            <property name="text">
             <string>...</string>
            </property>
            <property name="popupMode">
             <enum>QToolButton::InstantPopup</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="pteSessionStatus">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>50</height>
          </size>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QgsAuthSslConfigWidget" name="wdgtSslConfig" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>5</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Close|QDialogButtonBox::Save</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsAuthSslConfigWidget</class>
   <extends>QWidget</extends>
   <header>qgsauthsslconfigwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsCollapsibleGroupBoxBasic</class>
   <extends>QGroupBox</extends>
   <header>qgscollapsiblegroupbox.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>radioServerImport</tabstop>
  <tabstop>leServer</tabstop>
  <tabstop>spinbxPort</tabstop>
  <tabstop>btnConnect</tabstop>
  <tabstop>spinbxTimeout</tabstop>
  <tabstop>radioFileImport</tabstop>
  <tabstop>leCertPath</tabstop>
  <tabstop>btnCertPath</tabstop>
  <tabstop>pteSessionStatus</tabstop>
 </tabstops>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="btngrpImport"/>
 </buttongroups>
</ui>
